pytest_inmanta/plugin.py

```diff
@@ -247,14 +255,56 @@ def __init__(self, uri):
         self._env_id = cfg_env.get()
 
 
+class InmantaPluginsImportLoader:
+    """
+        Makes inmanta_plugins packages (Python source for inmanta modules) available dynamically so that tests can use them
+        safely without having to refresh imports when the compiler is reset.
+    """
+    def __init__(self, importer: "InmantaPluginsImporter") -> None:
+        self._importer: InmantaPluginsImporter = importer
+
+    def __getattr__(self, name: str):
+        submodules: Optional[Dict[str, ModuleType]] = self._importer.get_submodules(name)
+        fq_mod_name: str = f"inmanta_plugins.{name}"
+        if submodules is None or fq_mod_name not in submodules:
+            raise AttributeError("No inmanta module named %s" % name)
+        return submodules[fq_mod_name]
+
+
+class InmantaPluginsImporter:
+    def __init__(self, project: "Project") -> None:
+        self.project: Project = project
+        self.loader: InmantaPluginsImportLoader = InmantaPluginsImportLoader(self)
+        # expose module imports from self.modules.some_module
+        self.modules: InmantaPluginsImportLoader = self.loader
```

Comment:
Are these variables used anywhere?

Reply:
Indeed they aren't. `self.loader` should be though. The other I forgot to remove. Fixed both.